File: scala/gigasecond/src/main/scala/Gigasecond.scala
import java.time.temporal.ChronoUnit
import java.time.{LocalDate, LocalDateTime, LocalTime}

/**
  * Created by mz on 3/18 018.
  */
object Gigasecond {

  def addGigaseconds(datetime: LocalDateTime): LocalDateTime =
    datetime.plus(1000000000L, ChronoUnit.SECONDS)

  def addGigaseconds(date: LocalDate): LocalDateTime =
    addGigaseconds(LocalDateTime.of(date, LocalTime.MIN))
}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

/**
 * Created On 05/29 2017
 *
 * @author hzpengjunjian@corp.netease.com
 */
class Gigasecond(age: LocalDateTime) {

    constructor(age: LocalDate) : this(age.atStartOfDay())

    val date: LocalDateTime = age.plusSeconds(1000000000)
}